### PR TITLE
Add quantize option to the coreml script

### PR DESCRIPTION
**Pull request by cccclai**
Summary:
Add the quantizer to the export script. The commandline is
```
python executorch.examples.apple.coreml.scripts.export -m resnet18 --quantize
```

Differential Revision: D63476611

### CODE REVIEW

**Comment by cccclai** [comment]
@cymbalrush @YifanShenSZ we observed the latency number is quite when loading the quantized model. Can you share some guidance on how to have the optimum latency on the `resnet` flow here?

**Review by YifanShenSZ — APPROVED** [review]

**Review by shoumikhin — APPROVED** [review]

File: examples/apple/coreml/scripts/export.py [hunk 1]
```diff
@@ -110,9 +123,10 @@ def export_lowered_module_to_executorch_program(lowered_module, example_inputs):
     return exec_prog
 
 
-def save_executorch_program(exec_prog, model_name, compute_unit):
+def save_executorch_program(exec_prog, model_name, compute_unit, quantize):
     buffer = exec_prog.buffer
-    filename = f"{model_name}_coreml_{compute_unit}.pte"
+    data_type = "quantize" if quantize else "fp"
+    filename = f"{model_name}_coreml_{compute_unit}_{data_type}.pte"
```
Comment (huydhn) [inline examples/apple/coreml/scripts/export.py:129]: I think this line is responsible for the coreml failure in trunk https://hud.pytorch.org/pytorch/executorch/commit/77e7ad10903ed3bcf16e41d04a5e595ecaa1458b.  It changes the export model file name from `mv3_coreml_all.pte` to `mv3_coreml_all_fp.pte`.  The name is used in many different tests https://github.com/search?q=repo%3Apytorch%2Fexecutorch+mv3_coreml_all&type=code.

cc @SS-JIA @cccclai

Comment (huydhn) [reply examples/apple/coreml/scripts/export.py:129]: Trying to revert the change fixes the failures https://github.com/pytorch/executorch/pull/5906

Comment (cccclai) [reply examples/apple/coreml/scripts/export.py:129]: ah good finding. Sorry I didn't realize the name was hardcoded. Approve the revert change.

**Pull request closed without merging**